module/wa/module.py

```diff
@@ -967,18 +968,44 @@ def import_wa(i):
             pname=None
             imported_params={}
 
-            for name, obj in inspect.getmembers(cs):
-                if name!='AndroidUxPerfWorkload' and inspect.isclass(obj):
-                    try:
-                        addr=getattr(cs, name)
-                        pp=addr.parameters
-                        pname=addr.package
-                    except Exception as e:
-                        pass
-
-                    if pp!=None:
-                        wa_name=name
-                        break
+            # Only examine classes
+            for name, obj in inspect.getmembers(cs, predicate=inspect.isclass):
```

Comment:
Does that require wa to be run as an import rather than from a shell? I believe CK runs wa as a command.

Otherwise think that requires shifting the logic from 'on run of workload' into the package installation script as (as far as I know) wa is invoked via shell rather than by direct python call. One requires an extra op per-run, the other an extra install of a package after CK update.

I don't have a strong feeling either way.